### PR TITLE
Istio metrics

### PR DESCRIPTION
**Pull request by orangedeng**
This PR support Istio metric graph.

1. Get Istio metric graph through API
2. Create Istio metric expression after enabling Istio
3. Some integration test about it

### CODE REVIEW

**Comment by loganhz** [comment]
LGTM https://github.com/rancher/rancher/pull/20234#issuecomment-495586018

File: tests/integration/suite/test_app.py [hunk 1]
```diff
@@ -86,6 +87,69 @@ def test_app_wordpress(admin_pc, admin_mc):
     wait_for_workload(client, ns.name, count=2)
 
 
+@pytest.mark.skip(reason="istio disabled")
```
Comment (cjellick) [inline tests/integration/suite/test_app.py]: A disabled test does not have much use. If this cannot be legitimately tested as an integration test, then remove it.

**Comment by cjellick** [comment]
I believe we agreed that Istio would be just an app deployment, completely independent of Rancher. This tightly couples the app deployment to backend rancher code. It creates a dependency between Rancher and Istio that we had hoped to avoid. 

We may need to revisit this design.

File: pkg/controllers/user/istio/metricExpression.go [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+		return err
+	}
+	if err := json.Unmarshal(jsondata, obj); err != nil {
+		return err
+	}
+	return nil
+}
+
+var (
+	IstioMetricsTemplate = `
```
Comment (cjellick) [inline pkg/controllers/user/istio/metricExpression.go]: this seems weird, next week, id like to discuss what it is and why its needed, but i can merge as-is, not a blocker

Comment (loganhz) [reply pkg/controllers/user/istio/metricExpression.go]: got it

**Comment by loganhz** [comment]
Types updated

**Comment by cjellick** [comment]
lgtm

**Pull request merged**